internal/wasm/module.go

```diff
@@ -48,6 +48,9 @@ type Module struct {
 	ImportGlobalCount,
 	ImportMemoryCount,
 	ImportTableCount Index
+	// ImportModuleNames hold the module names imported by this module. This is used during instantiation to mitigate
+	// the lock contention.
+	ImportModuleNames map[string]struct{}
```

Comment:
It looks like this is only iterated on, maybe make it a `[]string`, could even be sorted for more determinism.

Reply:
I thought about it first, but in anyway, we need to duplicate at binary decoding phrase to create the slice. So I decided to reuse the map here. But not a strong opinion. In fact, somehow current change seems not to reduce the allocation at all according to the perf result 😞 maybe something wrong here

Reply:
using the slice resulted in the same.. 🤷